Tasks/Common/azurerest-common/azurerestutility.ts

```diff
@@ -995,22 +999,71 @@ export async function createAppInsightsWebTest(endpoint, appInsightsResourceGrou
     webTestData.tags[webTestHiddenLink] = "Resource";
 
     var metadataUrl = endpoint.url + 'subscriptions/' + endpoint.subscriptionId + '/resourceGroups/' + appInsightsResourceGroupName +
-            '/providers/microsoft.insights/webTests/' + webTestName + '?api-version=2015-05-01';
+        '/providers/microsoft.insights/webTests/' + webTestName + '?api-version=2015-05-01';
 
     tl.debug('Updating Application insights resources : ' + metadataUrl);
-    let options: rm.IRequestOptions = {};
+    let options: IRequestOptions = {};
     options.additionalHeaders = headers;
-    rc.replace(metadataUrl, webTestData, options).then((response) => {
-        if(response.statusCode === 200) {
+    restClient.replace(metadataUrl, webTestData, options).then((response) => {
+        if (response.statusCode === 200) {
             deferred.resolve(response.result);
         } else {
             tl.debug("Action: createAppInsightsWebTest, Response: " + JSON.stringify(response));
             deferred.reject(response.statusCode);
         }
     },
-    (error) => {
-        deferred.reject(error);
-    });
+        (error) => {
+            deferred.reject(error);
+        });
+
+    return deferred.promise;
+}
+
+/**
+ * Adds release annotation to an application insights resource
+ * 
+ * @param   endpoint                Service Principal Endpoint
+ * @param   appInsightsResourceId   ResourceId of the application insights resource
+ * @param   isDeploymentSuccess     Add annotation for success or failed deployment
+ */
+export async function addReleaseAnnotation(endpoint, appInsightsResourceId: string, isDeploymentSuccess: string): Promise<any> {
+    let deferred: Q.Deferred<any> = Q.defer<any>();
+
+    let accessToken = await getAuthorizationToken(endpoint);
+    let restOptions: IRequestOptions = {
+        additionalHeaders: {
+            "Authorization": `Bearer ${accessToken}`
+        }
+    };
+
+    let restUrl = `${endpoint.url}${appInsightsResourceId}/Annotations?api-version=2015-05-01`;
+    let releaseAnnotationProperties: IDictionaryStringTo<string> = {
+        "Label": isDeploymentSuccess ? "Success" : "Error", // Label decides the icon for release annotation
+        "Deployment Uri": Utils.getDeploymentUri()
+    };
+
+    let annotationName = tl.getVariable("Release.ReleaseName") || `${tl.getVariable("Build.DefinitionName")} - ${tl.getVariable("Build.BuildNumber")}`;
+    let releaseAnnotation = {
+        "AnnotationName": annotationName,
+        "Category": "Text",
```

Comment:
Should we change the category to ["Deployment"](https://github.com/Microsoft/ApplicationInsights-Home/blob/master/API/CreateReleaseAnnotation.ps1#L107)?